plugins/integration/providers/cielo24/Cielo24ClientHelper.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * @package plugins.cielo24
+ */
+class Cielo24ClientHelper
+{
+	const API_CALL_NAME_PATTERN = "{API-CALL-NAME}";
+	
+	private $supportedLanguages = array();
+	private $baseEndpointUrl = null;
+	
+	public function __construct($username, $password)
+	{
+		$cielo24ParamsMap = kConf::get('cielo24','integration');
+		$baseUrl = $cielo24ParamsMap['base_url'];
+		$version = $cielo24ParamsMap['version'];
+		$baseUrl = $baseUrl . "account/login?v=$version";
+		$loginParams = array("username" => $username, "password" => $password);	
+		$loginUrl = $this->addUrlParams($baseUrl, $loginParams);
+	
+		$loginResult = self::sendAPICall($loginUrl);
+		$apiToken = $loginResult->ApiToken;
+			
+		$this->baseEndpointUrl = str_replace("account/login", self::API_CALL_NAME_PATTERN, $baseUrl);
```

Comment:
@shpiNi why do you need `API_CALL_NAME_PATTERN`? it seems like an unneeded string transformation that you are doing, you can write the code better to avoid it.
Always keep the `$this->baseEndpointUrl` untouched.